includes/Modules/Analytics_4.php

```diff
@@ -831,4 +863,455 @@ public function check_service_entity_access() {
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array|null $orderby Data request orderby value.
+	 * @return Google_Service_AnalyticsData_OrderBy[] An array of AnalyticsData OrderBy objects.
+	 */
+	protected function parse_reporting_orderby( $orderby ) {
+		if ( empty( $orderby ) || ! is_array( $orderby ) ) {
+			return array();
+		}
+
+		$results = array_map(
+			function ( $order_def ) {
+				$order_def = array_merge(
+					array(
+						'fieldName' => '',
+						'sortOrder' => '',
+					),
+					(array) $order_def
+				);
+
+				if ( empty( $order_def['fieldName'] ) || empty( $order_def['sortOrder'] ) ) {
+					return null;
+				}
+
+				$metric_order_by = new Google_Service_AnalyticsData_MetricOrderBy();
+				$metric_order_by->setMetricName( $order_def['fieldName'] );
+				$order_by = new Google_Service_AnalyticsData_OrderBy();
+				$order_by->setMetric( $metric_order_by );
+				$order_by->setDesc( 'DESCENDING' === $order_def['sortOrder'] );
+
+				return $order_by;
+			},
+			// When just object is passed we need to convert it to an array of objects.
+			wp_is_numeric_array( $orderby ) ? $orderby : array( $orderby )
+		);
+
+		$results = array_filter( $results );
+		$results = array_values( $results );
+
+		return $results;
+	}
+
+	/**
+	 * Creates a new Analytics 4 site request for the current site and given arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $property_id Analytics 4 property ID.
+	 * @param array  $args {
+	 *     Optional. Additional arguments.
+	 *
+	 *     @type array                                           $dimensions        List of request dimensions. Default empty array.
+	 *     @type Google_Service_AnalyticsData_FilterExpression[] $dimension_filters List of dimension filter instances for the specified request dimensions. Default empty array.
+	 *     @type string                                          $start_date        Start date in 'Y-m-d' format. Default empty string.
+	 *     @type string                                          $end_date          End date in 'Y-m-d' format. Default empty string.
+	 *     @type string                                          $page              Specific page URL to filter by. Default empty string.
+	 *     @type int                                             $row_limit         Limit of rows to return. Default empty string.
+	 * }
+	 * @return Google_Service_AnalyticsData_RunReportRequest|WP_Error Analytics 4 site request instance.
+	 */
+	protected function create_analytics_site_data_request( $property_id, array $args = array() ) {
+		$args = wp_parse_args(
+			$args,
+			array(
+				'dimensions'        => array(),
+				'dimension_filters' => array(),
+				'start_date'        => '',
+				'end_date'          => '',
+				'page'              => '',
+				'row_limit'         => '',
+			)
+		);
+
+		$request = new Google_Service_AnalyticsData_RunReportRequest();
+		$request->setProperty( self::normalize_property_id( $property_id ) );
```

Comment:
This seems to be unnecessary but we can look at it in a separate issue as I don't think it's hurting anything here. The property ID is provided as a path parameter in the API request, but the object we're building here is essentially the body of that request so it's being duplicated. It's also not mentioned in the [schema for the request body](https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/properties/runReport), but is specifiable in the API explorer 😄 🤷‍♂️ 

Perhaps this is a leftover detail from an earlier version of the API or maybe there for consistency with the batch method (although I think a batch is still limited to the same property). See https://github.com/googleapis/google-api-php-client-services/blob/ff7e67045c666d292e07e0898ce63be458592745/src/AnalyticsData/Resource/Properties.php#L199-L205

Reply:
Yes, possibly unnecessary, worth taking another look at :+1: